controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -296,44 +296,48 @@ func (r *ROSAControlPlaneReconciler) reconcileNormal(ctx context.Context, rosaSc
 		AWSCreator:     creator,
 	}
 
-	_, machineCIDR, err := net.ParseCIDR(rosaScope.ControlPlane.Spec.Network.MachineCIDR)
-	if err == nil {
-		spec.MachineCIDR = *machineCIDR
-	} else {
-		// TODO: expose in status
-		rosaScope.Error(err, "rosacontrolplane.spec.network.machineCIDR invalid", rosaScope.ControlPlane.Spec.Network.MachineCIDR)
-		return ctrl.Result{}, nil
-	}
+	if networkSpec := rosaScope.ControlPlane.Spec.Network; networkSpec != nil {
+		if networkSpec.MachineCIDR != "" {
+			_, machineCIDR, err := net.ParseCIDR(networkSpec.MachineCIDR)
+			if err != nil {
+				// TODO: expose in status
+				rosaScope.Error(err, "rosacontrolplane.spec.network.machineCIDR invalid", networkSpec.MachineCIDR)
+				return ctrl.Result{}, nil
```

Comment:
Could we move this to a validation webhook, and just return the error here (backoff loop) or set a condition?

Reply:
Returning an error when the data provided by the user is incorrect would only lead to infinite retry by the controller and not result in a good outcome. It seems universally correct to expose states like this in status so that the user knows that they need to update the object's `.spec` and so that the controller does not do extra reconciliation loops that cannot possibly result in a better outcome.